lib/mergeDuplicateMaterials.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+var Cesium = require('cesium');
+var deepEqual = require('deep-equal');
+
+var defined = Cesium.defined;
+
+var removeUnusedMaterials = require('./removeUnusedMaterials');
+
+module.exports = mergeDuplicateMaterials;
+
+function mergeDuplicateMaterials(gltf) {
+    var materials = gltf.materials;
+    var materialIdMapping = {};
+    var uniqueMaterials = [];
+    for (var materialId in materials) {
+        if (materials.hasOwnProperty(materialId)) {
+            var material = materials[materialId];
+            var uniqueMaterialsLength = uniqueMaterials.length;
+            var unique = true;
+            for (var i = 0; i < uniqueMaterialsLength; i++) {
+                var uniqueMaterialId = uniqueMaterials[i];
+                var uniqueMaterial = materials[uniqueMaterialId];
+                if (deepEqual(material, uniqueMaterial)) {
```

Comment:
If two materials are the same but have different names, should they also get merged?